### DISCUSSION TITLE
User interface. If a keyboard shortcut exists, there should be a button for it.

### DISCUSSION THREAD

**Discussion by TheGoklayeh** [Orama-Interactive/Pixelorama#575] [Ideas]
Shortcuts by definition are a faster alternative to a more obvious (but slower) way. 
Otherwise it's not a shortcut, it's the only path.

When features are only doable through shortcuts, they are hidden and way harder to find. 
This is a problem, specially for important feature, as many people will just assume they are missing.

It's OK to have more icons. Specially for things like _Lines_, _Moving Selection_, making shapes, etc.
Graphicsgale has a buch of them, as does Aseprite.

Having your icons along the vertical side and the options of the tool selected horizontally might be a nice setup.

You could even allow people to hide the task bars they don't use if they want.

**Comment by hsandt** [comment]
I have the opposite issue where I can't find a shortcut for a visible button or option, so I'd also suggest the other way round: each button/option should have a customizable shortcut (could be empty by default).

For instance I'm trying to assign a shortcut to Toggle Layer Visibility but unlike menu items those are very hard to find in the action list. Until Action search by name (https://github.com/Orama-Interactive/Pixelorama/discussions/983) is implemented it will also be hard to tell if a shortcut exists or not.

To sum up, it would be nice to have an equivalence between actions listed in Preferences > Shortcuts and menu items + buttons + options (for options it's a bit trickier: toggle flag is easy, but for numeric fields you'd need increase/decrease shortcuts with some amount, and for dropdown select next/previous entry). And the action should be relatively easy to find in the hierarchy following a strict logic (e.g. Specific Tool Name > Options).

P.S.: I wonder what actions have shortcuts but no buttons, can you give some examples?

**Comment by Variable-ind** [comment]
> Shortcuts by definition are a faster alternative to a more obvious (but slower) way. Otherwise it's not a shortcut, it's the only path.
> 
> When features are only doable through shortcuts, they are hidden and way harder to find. This is a problem, specially for important feature, as many people will just assume they are missing.
> 
> It's OK to have more icons. Specially for things like _Lines_, _Moving Selection_, making shapes, etc. Graphicsgale has a buch of them, as does Aseprite.
> 
> Having your icons along the vertical side and the options of the tool selected horizontally might be a nice setup.
> 
> You could even allow people to hide the task bars they don't use if they want.

I think every shortcut has a corresponding button if I remember correctly. Could you tell me about the shortcuts more 😁?
